### ISSUE TITLE
[VL] add div function

### ISSUE THREAD

**Issue by guixiaowen** [apache/incubator-gluten#5574]
### Description

div function
expr1 div expr2 - Divide expr1 by expr2. It returns NULL if an operand is NULL or expr2 is 0. The result is casted to long.

Examples:

`
SELECT 3 div 2;
SELECT INTERVAL '1-1' YEAR TO MONTH div INTERVAL '-1' MONTH;
 -13

`

The implementation class of div is: IntegralDivide.

In 3.2.x, supports two types of div:

eg:  long div long ,result is long

**Comment by guixiaowen** [comment]
@PHILO-HE PH hi， Can you help me check if the direction of the issuse I mentioned is correct.

**Comment by guixiaowen** [comment]
@Yohahaha yo hi, Can you help me check if the direction of the issuse I mentioned is correct.

**Comment by Yohahaha** [comment]
> @Yohahaha yo hi, Can you help me check if the direction of the issuse I mentioned is correct.

yeah, IntegralDivide for Long type is missed, you can verify it with an UT.

**Comment by guixiaowen** [comment]
> > @Yohahaha yo hi, Can you help me check if the direction of the issuse I mentioned is correct.
> 
> yeah, IntegralDivide for Long type is missed, you can verify it with an UT.

@Yohahaha  3q.

**Comment by PHILO-HE** [comment]
@guixiaowen, the proposal looks good! Just left some comments on the velox PR. Thanks!

**Comment by guixiaowen** [comment]
> @guixiaowen, the proposal looks good! Just left some comments on the velox PR. Thanks!

OK. Thanks.
